2022/incoming/create-your-excel-using-perl.pod

```diff
@@ -0,0 +1,170 @@
+		      Scandinavia => [ 'Pepparkakor', 'Ruiskakut', 'Lussekatter', 'Krumkake' ],
+		      Netherlands => [ 'Spekulaas', 'Kerstkranjes'],
+		      Germany => [ 'Lebkuchen', 'Stollen' ],
+		      Switzerland => [ 'Basler Läckerli', 'Brunsli' ],
+		      Italy => [ 'Panettone', 'Pignoli', 'Cannoli' ],
+		      Spain => [ 'Polvorones', 'Mantecados', 'Turrón' ],
+		      Greece => [ 'Melomakarona' ],
+		      };
+		      
+	#Sort the countries alphabetically
+	my @countries = sort keys %$cookies;
+
+	#Add the countries to the baking sheet, one country per column.
+	$baking_sheet->set_column( 0, scalar keys %$cookies, 15);
+	$baking_sheet->write( 'A1', \@countries, $format2 );
+
+	my $i = 0;
+	for ( sort keys %$cookies ) {
+	  #Place Your cookies to the baking sheet
+	  $baking_sheet->write( 1, $i, [ $cookies->{$_} ], $format3 );
+	  $i++;
+	}
+
+	#write some statistics about how many types of cookies we have per country
+	$stats->write( 'A1', [\@countries], $format2 );
+
+	#Reset the counter
+	$i = 0;
+	#count how many cookies we country wise.
+	for ( sort keys %$cookies ) {
+	  #Place Your cookies to the baking sheet
+	  $stats->write( $i, 1, scalar @{$cookies->{$_}} , $format3 );
+	  $i++;
+	}
+        
+=end code
+
+Many things are happening here.
+
+First we get a list of countries sorted alphabetically.
+Second we are writing those countries into the first row, alongside some formats, defined above.
+Then comes the first usage of C<write()>, where we are using the A1 notation to start from the top left corner, then comes the reference to the C<@countries> array, followed by the format2.
+When we pass an array reference to C<write()>, it will write the elements of the array in the same row. It is equivalent to write_row().
+If You pass and array ref of array refs to C<write()>, it is the same if You would call C<write_col()>, it will write to the columns, as demonstrated in the lines 22 and 27.
+
+For more on C<write()>, please check the L<write section|https://metacpan.org/pod/Excel::Writer::XLSX#write(-$row,-$column,-$token,-$format-)> on the module's page.
+
+=head2 Outro
+
+Santa was cheerful about this guide, especially that it contains the long avaited Christmas cookie statistics. 
+And perhaps the next task will be to extend this report with more Christmas cookies from around the world.
+
+We just scratched the surface of C<Excel::Writer::XLSX> with this short tutorial. You can find a comprehensive guide on CPAN, L<Excel::Writer::XLSX|https://metacpan.org/pod/Excel::Writer::XLSX>.
+This is a very useful and powerful module, which can deal with a range of tasks from the basics of generating reports to the heavy lifting of the work with VBA in Excel.
+To cover all of it's features would take a whole book.
+
+You can find the whole script on my Github L<cookies_test.pl|https://github.com/x1mandi/padvent_code/blob/main/cookies_test.pl>, so that You can test it on Your own and play with the module. \
```

Comment:
The backslash at the end can be removed.

Reply:
Done